### DISCUSSION TITLE
Extracting pairwise forces from LAMMPS.

### DISCUSSION THREAD

**Discussion by rohskopf** [deepmodeling/deepmd-kit#211] [Q&A]
How can we extract the pairwise forces F_ij between atoms i and j in the LAMMPS interface? Is that information stored anywhere in the force calculation?

I realize there are 3-body terms F_ijk, can we extract those as well?

**Comment by amcadmus** [comment]
Deep potential is a multi-body potential, and it is not straightforward to extract 2-body, 3-body components out of it.

**Comment by rohskopf** [comment]
> Deep potential is a multi-body potential, and it is not straightforward to extract 2-body, 3-body components out of it.

If you look at the documentation, the total force on an atom is literally a sum over neighbors. The components of those sums must be stored somewhere.

**Comment by amcadmus** [comment]
> > Deep potential is a multi-body potential, and it is not straightforward to extract 2-body, 3-body components out of it.
> 
> If you look at the documentation, the total force on an atom is literally a sum over neighbors. The components of those sums must be stored somewhere.

Yes I agree, the total force is a sum of neighbors because the atomic energy is a function of relative position of neighbor atoms. It is noted that each component depends not only on the position of one neighbor, but also on the positions of all other neighbors (actually also on neighbors of a neighbor). That's why DP is a multi-body interaction. 

In deepmd-kit, only the total force is stored. The components are discarded after they are added to the total force.  If you really need them you have to hack the code in source/op/prod_force***.cc

**Comment by K1ngRat1027** [comment]
hello rohskopf, have you solved this problem? I have similar question too and really would like to seek your help : )
